nogotofail/mitm/connection/handlers/data/ssl.py

```diff
@@ -203,3 +203,25 @@ def on_ssl(self, client_hello):
                 self.log(logging.ERROR,
                         "Client enabled SSLv3 protocol without TLS_FALLBACK_SCSV")
             self.log_attack_event(data="SSLv3")
+
+
+@handler.passive(handlers)
+class ClientCompressionDetectionHandler(DataHandler):
+    name = "compressiondetection"
+    description = "Detect insecure non-null compression methods in TLS Client Hellos"
+
+    def _handle_compression_detected(self, compression_methods, message):
+            self.log(logging.ERROR, message)
+            self.log_attack_event(data=compression_methods)
+            self.connection.vuln_notify(vuln.VULN_TLS_CLIENT_COMPRESSION)
```

Comment:
Will this cause the test to exit immediately? Not clear why you need to print all compression methods when you're already iterating over them anyway.

Reply:
Yes it will/does. I am trying to keep the same convention used throughout in rest of the handlers. The thinking seems - dumping all compression methods in the event log in case anything offending is found can be helpful in diagnosing (like you want to see which call non-null compression methods client is sending in this case). Need to iterate thru to be able to gather them in a string.